logic.py

```diff
@@ -215,3 +216,50 @@ def get_next_element(handshake_url, request):
     except IndexError:
         # An index error will occur here when the workflow is complete
         return None
+
+
+def save_galley(article, request, uploaded_file, is_galley, label=None, save_to_disk=True):
+    new_file = files.save_file_to_article(uploaded_file, article, request.user, save=save_to_disk)
+    new_file.is_galley = is_galley
+    new_file.label = label
+
+    new_file.save()
+    article.save()
+
+    _type = None
+
+    if new_file.mime_type in files.HTML_MIMETYPES:
+        _type = 'html'
+        if not label:
+            label = 'HTML'
+
+        with open(new_file.self_article_path(), 'r+', encoding="utf-8") as f:
+            html_contents = f.read()
+            f.seek(0)
+            cleaned_html = remove_css_from_html(html_contents)
+            f.write(cleaned_html)
+            f.truncate()
+
+    elif new_file.mime_type in files.XML_MIMETYPES:
+        _type = 'xml'
+        if not label:
+            label = 'XML'
+    elif new_file.mime_type in files.PDF_MIMETYPES:
+        _type = 'pdf'
+        if not label:
+            label = 'PDF'
+
+    if not label:
+        label = 'Other'
+    if not _type:
+        _type = 'other'
+
+    new_galley = core_models.Galley.objects.create(
+        article=article,
+        file=new_file,
+        label=label,
+        type=_type,
+        sequence=article.get_next_galley_sequence(),
+    )
+
+    return new_galley
```

Comment:
why can't we make this change in the `production` app instead? sounds like it would be benefitial